src/main/kotlin/no/nav/medlemskap/services/aareg/AaRegClient.kt

```diff
@@ -80,9 +83,31 @@ class AaRegClient(
     private fun LocalDate.tilIsoFormat() = this.format(DateTimeFormatter.ISO_LOCAL_DATE)
 }
 
-class AaRegService(private val aaRegClient: AaRegClient) {
+class AaRegService(
+        private val aaRegClient: AaRegClient,
+        private val eregClient: EregClient,
+        private val pdlClient: PdlClient) {
 
-    suspend fun hentArbeidsforhold(fnr: String, callId: String, fraOgMed: LocalDate? = null, tilOgMed: LocalDate? = null) =
-            mapAaregResultat(aaRegClient.hentArbeidsforhold(fnr, callId, fraOgMed, tilOgMed))
+    suspend fun hentArbeidsforhold(fnr: String, callId: String, fraOgMed: LocalDate? = null, tilOgMed: LocalDate? = null): List<Arbeidsforhold> {
+        val arbeidsforhold = aaRegClient.hentArbeidsforhold(fnr, callId, fraOgMed, tilOgMed)
+        val arbeidsgiver: List<AaRegOpplysningspliktigArbeidsgiver> = arbeidsforhold.map { it.arbeidsgiver }
+        val arbeidsgiversLand = hentArbeidsgiversLand(arbeidsgiver)
 
+        return mapAaregResultat(arbeidsforhold, arbeidsgiversLand)
+    }
+
+    private suspend fun hentArbeidsgiversLand(opplysningspliktigArbeidsgiver: List<AaRegOpplysningspliktigArbeidsgiver>): Map<String, String> {
+        return opplysningspliktigArbeidsgiver.associateBy(
+                { arbeidsgiver ->
+                    arbeidsgiver.organisasjonsnummer ?: (arbeidsgiver.offentligIdent ?: (arbeidsgiver.aktoerId ?: ""))
```

Comment:
Her henter du orgnr eller offentligident eller aktorid, men nedenfor, når du gjør kall til PDL, så bruker du bare offentligident. Det bør vel helst være konsistent, tenker jeg?

Reply:
Den tomme strengen er jeg litt usikker på.. Det skal vel aldri skje at ingen av IDene er satt, men da kunne det kanskje være like greit å hive en exception i det tilfellet?

Reply:
Den tomme strengen var mest for å få det til å bygge. Uten den ville det være build error.
Burde vi stoppe hele systemet dersom en slik error antreffer? En exception vil ta ned hele systemet i dette tilfelle vil jeg tro

Reply:
En exception vil ta kverken på den requesten, ja. Vi har diskutert det litt før, hva gjør vi hvis et av baksystemene ikke svarer etc. Generelt har vi da landa på at de er greit å feile (men at vi bør ha lit retry-mekanismer på plass og sånt) Om det er annerledes her... Tja. Jo, det er det nok kanskje. Dette er jo ikke en hickup, en temporær feil. Så det hjelper ikke å prøve igjen om ti minutter. Så sånn sett så bør vi vel håndtere det litt annerledes. Nå er vel dette først og fremst en hypotetisk problemstilling, håper jeg. Ellers har AaReg elendig datakvalitet. Men hvis det skulle inntreffe, hva bør skje da? Vi kan anta at arbeidsgiver er norsk? Vi kan sende personen til manuell behandling? Jeg ville laget et avklaringspunkt på dette som vi kan diskutere med Helle.